kernel/kernel-api/src/main/java/io/delta/kernel/Table.java

```diff
@@ -66,6 +71,15 @@ static Table forPath(Engine engine, String path) {
    */
   String getPath(Engine engine);
 
+  /**
+   * The catalog identifier of this {@link Table} instance.
+   *
+   * @param engine {@link Engine} instance.
+   * @return the table identifier, or {@link Optional#empty()} if none is set.
+   * @since 3.3.0
+   */
+  Optional<TableIdentifier> getTableIdentifier(Engine engine);
```

Comment:
I thought we wanted to remove the `Engine` as arg to APIs going forward.

Reply:
Yup! We do. We should do that in one final pass. Until then, I'm going to stick with the current convention.